Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -1213,7 +1213,7 @@ void RenderBlockFlow::determineLogicalLeftPositionForChild(RenderBox& child, App
     LayoutUnit startPosition = borderAndPaddingStart();
     LayoutUnit initialStartPosition = startPosition;
     if ((shouldPlaceVerticalScrollbarOnLeft() || style().scrollbarGutter().bothEdges) && isHorizontalWritingMode())
-        startPosition += (writingMode().isLogicalLeftInlineStart() ? 1 : -1) * verticalScrollbarWidth();
+        startPosition += (writingMode().isLogicalLeftInlineStart() ? 1 : -1) * verticalScrollbarWidthClampedToContentBox();
```

Comment:
`auto verticalScrollbarWidthClampedToContentBox = std::min(verticalScrollbarWidth(), std::max(0_lu, contentBoxLogicalWidth()))`
would probably just work instead of introducing such specific function on RenderBox. -but even if we did introduce it , shouldn't it be on `RenderBlockFlow`? (I am also wondering if there's a reason for not using `contentBoxLogicalWidth`)

Reply:
so then it probably has to be `logicalWidth() - borderAndPaddingLogicalWidth()`. It would be great to figure out why `contentBoxLogicalWidth()` fails though

Reply:
@alanbaradlay - Updated. Sorry it took some time to get back to this PR.